src/test/java/web/tests/FlipkartTest.java

```diff
@@ -168,19 +170,34 @@ public void testAddToCartFunctionality(String product) {
     }
 
     /**
-     *
-     * @param item :accepting shoes from json
+     * @param item     :accepting shoes from json
      * @param position :accepting position i.e to select 2nd position shoes
      */
     @ParameterizedTest
     @CsvFileSource(resources = "/testProductPageDesign.csv")
-    public void testProductPageDesign(String item,int position){
+    public void testProductPageDesign(String item, int position) {
         ProductPage productPage = new HomePage().popUpCancel().setShoes(item).searchShoes().OpenProductPage(position);
         productPage.scrollToBottom();
-        ArrayList<String> headers  = new ArrayList<>(Arrays.asList(map.get("headers").split(",")));
-        for(String headersInSelectionPage:headers){
+        ArrayList<String> headers = new ArrayList<>(Arrays.asList(map.get("headers").split(",")));
+        for (String headersInSelectionPage : headers) {
             softAssert.assertThat(productPage.isSelectionDisplayed(headersInSelectionPage)).
                     as("Json selected Headers is Not matching with selection page Headers").isTrue();
         }
     }
+
+    @Test
+    public void flipkartSocialMedia() {
+        HomePage homePage = new HomePage().popUpCancel();
+        SocialMediaPage socialMediaPage=null;
+        ArrayList<String> socialMediaLinks = new ArrayList<>(Arrays.asList(map.get("links")
+                .split(",")));
+        for (int i = 0; i < socialMediaLinks.size(); i++) {
+            socialMediaPage =homePage.clickLink(socialMediaLinks.get(i));
+            String url = socialMediaPage.getSocialMediaUrl();
+            logger.info(url);
+            softAssert.assertThat(url.contains(socialMediaLinks.get(i).toLowerCase())).
+                    as("links of social media are not matching with the contents in url");
```

Comment:
soft assert won't assert this until and unless you don't specify the result - isTrue() or isFalse()